portalnet/src/utp_controller.rs

```diff
@@ -15,33 +25,253 @@ use crate::discovery::UtpEnr;
+                );
+                return Err(OverlayRequestError::ContentNotFound {
+                    message:
+                        "Unable to locate content on the network: unable to establish utp conn"
+                            .to_string(),
+                    utp: true,
+                    trace,
+                });
+            }
+        };
+
+        let mut data = vec![];
+        if let Err(err) = stream.read_to_eof(&mut data).await {
+            self.metrics
+                .report_utp_outcome(UtpDirectionLabel::Inbound, UtpOutcomeLabel::FailedDataTx);
+            debug!(%err, cid.send, cid.recv, peer = ?cid.peer.client(), "error reading data from uTP stream, while handling a FindContent request.");
+            return Err(OverlayRequestError::ContentNotFound {
+                message:
+                    "Unable to locate content on the network: error reading data from utp stream"
+                        .to_string(),
+                utp: true,
+                trace,
+            });
+        }
+
+        // report utp tx as successful, even if we go on to fail to process the
+        // payload
+        self.metrics
+            .report_utp_outcome(UtpDirectionLabel::Inbound, UtpOutcomeLabel::Success);
+        Ok(data)
+    }
+
+    /// Connect with a peer given the connection id, and transfer the data to the peer.
+    pub async fn connect_outbound_stream(&self, cid: ConnectionId<UtpEnr>, data: Vec<u8>) -> bool {
+        self.metrics
+            .report_utp_active_inc(UtpDirectionLabel::Outbound);
+        let stream = match self.connect_with_cid(cid.clone(), *UTP_CONN_CFG).await {
+            Ok(stream) => stream,
+            Err(err) => {
+                self.metrics.report_utp_outcome(
+                    UtpDirectionLabel::Outbound,
+                    UtpOutcomeLabel::FailedConnection,
+                );
+                debug!(
+                    %err,
+                    cid.send,
+                    cid.recv,
+                    peer = ?cid.peer.client(),
+                    "Unable to establish uTP conn based on Accept",
+                );
+                return false;
+            }
+        };
+
+        // send_utp_content handles metrics reporting of successful & failed txs
+        match Self::send_utp_content(stream, &data, self.metrics.clone()).await {
+            Ok(_) => true,
+            Err(err) => {
+                debug!(
+                    %err,
+                    %cid.send,
+                    %cid.recv,
+                    peer = ?cid.peer.client(),
+                    "Error sending content over uTP, in response to ACCEPT"
+                );
+                false
+            }
+        }
+    }
+
+    /// Accept an outbound stream given the connection id, and transfer the data to the peer.
+    pub async fn accept_outbound_stream(
+        &self,
+        cid: ConnectionId<UtpEnr>,
+        content_id: [u8; 32],
```

Comment:
```
        content_id: [u8; 32],
```
is only used in the debug log. I think we should just remove it in that case. As I think think a debug log justifies passing a variable

Reply:
> As I think think a debug log justifies passing a variable

? you do think or you don't think?

Reply:
Doesn't justify